### DISCUSSION TITLE
Layout that just hides the right sidebar

### DISCUSSION THREAD

**Discussion by DannyBen** [retypeapp/retype#89] [Ideas]
I was happy to see the `layout` page config option, since I wanted to hide the righthand side navigation, but quickly realized I am out of luck since such a layout does not exist... :)

In some cases - for example pages with a low number of headings, but with content that requires more width (screenshots, welcome page), it would be extremely useful to hide the inner page navigation, and utilize the full width of the page.

I hope such an additional layout option can be added in the future.

Also, as a side note, I found the option `layout: page` (== same as default, only hides from navigation) to be a bit odd, since:

1. It does not actually change the layout.
2. It is redundant with the `hidden: true` option.

**Comment by fabriciomurta** [comment]
> Also, as a side note, I found the option `layout: page` (== same as default, only hides from navigation) to be a bit odd, since:
> 
> 1. It does not actually change the layout.
> 2. It is redundant with the `hidden: true` option.

About this part, well, there's a little more to hiding a page than meets the eye. We are still working our way thru them, but when we want to hide a page we want to either include or exclude them from searches, tag listing, and probably other constraints. We are working on a way to better give control over visibility of pages, and you are going to see news about that soon. Hopefully to the better!

As for the layout it sounds pretty reasonable to add two alternatives to "central" selecting either right or left sidebar.

**Reply by DannyBen** [reply]
Thanks.

I was mainly referring to pages like [this](https://bashly.dannyb.co/):
![image](https://user-images.githubusercontent.com/2405099/126810695-4cfe3d05-fb81-4514-bb88-58357c9b1b7b.png)

- This is a "Welcome" page, which is expected to be a little more "celebratory"
- There is no need for me - on this particular page - to have a table of contents for the page
- The main content - containing an animated screencast - ends up being too small to read on some screens.

Of course - completely not a big deal, this little annoyance pales in comparison to the beauty that is Retype.

**Comment by geoffreymcgill** [comment]
Yes, we will be adding an option to hide the right (and left) sidebar columns. Hopefully in `v1.9.0`.

I'm not sure if adjusting the main content width will be supported. That's a delicate subject and one that comes with a few strongly held opinions. We will keep it in mind and discuss. 

Have you seen the Image [alignment options](https://retype.com/components/image/#alignment-options)? 

The `*plus` alignment options, such as `leftplus`, `rightplus`, and `centerplus` will only work if the page is configured with `layout: central` or `layout: blog`. 

Your [Bashly](https://bashly.dannyb.co/) home page would need to be configured with `layout: central`, but that would also hide the left navigation column on the home page.

Then you would configure the screencast svg as:

```
// Existing
[![Bashly Demo](/assets/cast.svg)](/demo/)

// Revised
--[![Bashly Demo](/assets/cast.svg)](/demo/)--
```

Maybe we could make `rightplus` work if the right column was hidden, and `leftplus` if the left column was hidden. Might be possible.

**Reply by DannyBen** [reply]
Thanks. The image alignment is a nice feature - I was unaware of it.

A few observations about it:

1. It does not work if the image is a link (your `// Revised` example)
2. It seems like the [docs](https://retype.com/components/image/#alignment-options) are wrong, to do `centerplus` I needed to do `--[image]--` and not `-[image]-`

**Reply by geoffreymcgill** [reply]
Right you are on both items. I apologize for the mistake. 

> It does not work if the image is a link (your // Revised example)

I forgot about the alignment only applying directly to the image component. We had discussed applying the same technique to many other components, but haven't got to that yet. I'm not sure how handy it would be. Wrapping an image in a link is a very common scenario, so we should support.

> It seems like the docs are wrong, to do centerplus I needed to do --[image]-- and not -[image]-

Thank you for pointing this out. ~~I will correct the docs.~~ [FIXED](https://retype.com/components/image/#alignment-options).

**Reply by 0xCooki** [reply]
please do add this layout functionality, thank you!
